File: Dataset/Leetcode/train/53/554.java
 class Solution {
    public int XXX(int[] nums) {
        int sum = 0;
        int ans = nums[0];

        for(int i = 0; i < nums.length; i++ ) {
/**           遍历子序列，和应该是逐渐增大的，如果遍历到总和小于0的情况，这个子序列就不是题需要的序列
 *            提问：为什么边界条件是总和小于0?
 */
            if(sum < 0)  
                sum = nums[i]; 
            else 
                sum += nums[i];  
            ans = Math.max(ans,sum);
        }
       
        return ans;
    }
}

File: Dataset/Leetcode/train/67/554.js
var XXX = function(a, b) {
    var re = new Array();
    var len = Math.max(a.length,b.length),flag = 0;
    
    for (var i = len-1 ; i >=0 ; i-- ) {
        // 当前位
        re[i] = (Number(a[i-len+a.length] ? a[i-len+a.length] : 0) + Number(b[i-len+b.length] ? b[i-len+b.length] : 0) + flag) % 2;
        // 模拟进位
        flag = Math.floor((Number(a[i-len+a.length] ? a[i-len+a.length] : 0) + Number(b[i-len+b.length] ? b[i-len+b.length] : 0) + flag) / 2);
    }
    // 首位判断是否进位，进位加一个1
    if (flag > 0 ) {
        re.unshift(1);
    }

    
    return re.join('')
};

